apps/backend/src/mutations/QuestionaryMutations.ts

```diff
@@ -53,9 +66,109 @@ export default class QuestionaryMutations {
     const questionIds: string[] = stepQuestions.map(
       (question) => question.question.id
     );
+
+    const genericTemplateQuestions = stepQuestions.filter(
+      (step) => step.question.dataType == DataType.GENERIC_TEMPLATE
+    );
+
+    if (genericTemplateQuestions) {
+      // Confirm dependency with dependency condition
+      const unsatisfiedDependencyQues = genericTemplateQuestions.filter(
+        (genericTemplateQues) => {
+          const notSatisfiedQuestions = genericTemplateQues.dependencies.filter(
+            (dependency) => {
+              const answer = answers.find(
+                (answer) => answer.questionId === dependency.dependencyId
+              );
+
+              if (answer) {
+                const { value } = JSON.parse(answer.value);
+                if (value[0] !== dependency.condition.params) {
+                  return true;
+                }
+              }
+            }
+          );
+
+          if (notSatisfiedQuestions.length != 0) {
+            return true;
+          }
+        }
+      );
+
+      unsatisfiedDependencyQues.forEach((subTemplateQues) => {
+        const config = subTemplateQues.config as SubTemplateConfig;
+        this.deleteSubTemplatesAnswers(
+          subTemplateQues.question.id,
+          config.templateId,
+          questionaryId,
+          agent
+        );
+      });
+    }
+
     await this.dataSource.deleteAnswers(questionaryId, questionIds);
   }
 
+  async deleteSubTemplatesAnswers(
+    questionId: string,
+    genericTemplateId: number | null,
+    questionaryId: number,
+    agent: UserWithRole | null
+  ): Promise<void> {
+    const proposal = (
+      await this.proposalDataSource.getProposals({
+        questionaryIds: [questionaryId],
+      })
+    ).proposals[0];
+
+    const genericTemplates =
+      await this.genericTemplateDataSource.getGenericTemplates(
+        {
+          filter: {
+            questionId: questionId,
+            proposalPk: proposal?.primaryKey,
+          },
+        },
+        agent
+      );
+
+    if (!genericTemplates || !genericTemplateId) {
+      return;
+    }
+
+    const templateSteps =
+      await this.templateDataSource.getTemplateSteps(genericTemplateId);
+    const genericTemplateStepQuestions = templateSteps.flatMap(
+      (step) => step.fields
+    );
+
+    if (genericTemplateStepQuestions === undefined) {
+      logger.logError('Expected to find sub template step, but was not found', {
+        genericTemplateId,
+        questionaryId,
+      });
+      throw new GraphQLError(
+        'Expected to find sub template step, but was not found'
+      );
+    }

```

Comment:
Maybe since this is a clean up if there is nothing to remove we can just return without throwing an error. But if it is really necessary maybe we can just log it.